src/openforms/registrations/api/views.py

```diff
@@ -56,6 +60,35 @@ def get_objects(self):
         return [ChoiceWrapper(choice) for choice in choices]
 
 
+@extend_schema_view(
+    get=extend_schema(
+        summary=_("List available Catalogi"),
+        parameters=[APIGroupQueryParamsSerializer],
+    ),
+)
+class CatalogiListView(ListMixin, APIView):
+    """
+    List the available Catalogi based on the provided API group.
+    """
+
+    authentication_classes = (authentication.SessionAuthentication,)
+    permission_classes = (permissions.IsAdminUser,)
+    serializer_class = CatalogusSerializer
+
+    def get_objects(self):
+        filter_serializer = APIGroupQueryParamsSerializer(
+            data=self.request.query_params
+        )
+        filter_serializer.is_valid(raise_exception=True)
+
+        client = filter_serializer.get_ztc_client()
+        if not client:
+            return []
```

Comment:
please address the missing test coverage 

Some more questions about this flow:

1. don't we enforce that either a ZGW API group or objects API group needs to be provided?
2. if the above is "no", what's stopping us from doing so?

It would be nicer if this code can just assume that the catalogi client will never be `None`. IMO this endpoint doesn't make sense to be called without a group/catalogi API reference and an HTTP 400 is more appropriate.